### PR TITLE
Rewrite of 2.0 to make it more testable

### PR DESCRIPTION
**Pull request by tduffield**
## Description

* Since we removed studio functionality, rename the package from
      ci-studio-common to ci-utils.
* Updated deploy and promotion through packages.chef.io
* Remove cross-internal pkg dependencies
  * Move CIStudio hook function into paths
  * Move sudo file modification into the only pkg that used it
* Rename 'paths' to 'install' and make it an interface for commands
* Add tests for most of the files
* Level up our usage of Cobra to make it easier to test
* Update GitHub Account to use GitHub App temporary tokens

Signed-off-by: Tom Duffield <tom@chef.io>
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tduffield** [comment]
Please do not merge this yet. There is an outstanding product rename that needs to get figured out.

File: cmd/ci-utils/commands/allow_unix.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+// addSudoPermission will ensure that the user can run the given command with sudo without a password
+func addSudoPermission(command string, user string) error {
+	sudoersFilePath := fmt.Sprintf("/etc/sudoers.d/%s", user)
+	newLine := fmt.Sprintf("%s ALL=NOPASSWD:SETENV: %s", user, command)
+
+	err := fs.AppendIfMissing(sudoersFilePath, []byte(newLine), 0440)
+	if err != nil {
+		return errors.Wrapf(err, "failed append sudo permission for %s to %s", user, sudoersFilePath)
```
Comment (scotthain) [inline cmd/ci-utils/commands/allow_unix.go]: failed `to` append sudo....

File: cmd/vault-util/commands/root.go [hunk 2]
```diff
@@ -1,65 +1,86 @@
...
 		log.Fatal(err)
 	}
 }
 
 func init() {
 	cobra.OnInitialize(initConfig)
 
-	// global config
-	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", fmt.Sprintf("configuration file (default is %s/vault-util.toml)", paths.SettingsDir))
+	viper.SetDefault("aws.region", "us-east-1")
```
Comment (scotthain) [inline cmd/vault-util/commands/root.go]: should these be configurable?

Comment (tduffield) [reply cmd/vault-util/commands/root.go]: yes, it is.

Comment (scotthain) [reply cmd/vault-util/commands/root.go]: Oh der - misread it!!

**Pull request merged**